### ISSUE TITLE
How to assign Coaches/Managers/Associates to Athletes for individual sports such as Boxing?

### ISSUE THREAD

**Issue by Pupix** [iptc/sport-schema#190]
Hello,
During the past week I've been looking at and using this schema as a reference for a project I am working on.

From what I am seeing there's been a lot of emphasis on team sports, as the `Membership` assumes that one `Individual` will be part of a `Team`.

I am unable to figure out how to assign a Coach/Manager/Associate to an Athlete with the current implementation. Perhaps I am reading it wrong.

A very real example would be Boxing, let's say **Mike Tyson** and his _Coach_ and _Manager_ **Cus D'Amato**. Yes there's an entire team of people and a whole business behind the scenes, but as far as the sport is concerned everything revolves around the `Athlete`.
Do I need to have both `Athlete` **Mike Tyson** and `Team` **Mike Tyson** or is there something missing in the schema?

Cheers!

Off topic:
What about `Teams` having `membershipOf` a `Club/Organization` (Which t doesn't exist currently) in things like Football/Soccer? And a `Club` that has multiple `Teams` in different `Competitions`, such as *Main team* and *Youth team* or something like that. Happens a lot in esports too, where an `Organization` has at least one team `Team` for every different `Game/Competition`.

**Comment by Pupix** [comment]
Hi Paul, thanks for getting back to me.

I am working on a sports management platform that has a customer/fan facing app and a backoffice/dashboard for the team/organization managers to do data entry. [Here's a landing page.](https://apiforge.it/sports)

We have done a first iteration, for a local basketball team, to validate the feasibility of the solution. Here's the customer facing side of it for [Android](https://play.google.com/store/apps/details?id=com.apiforge.vastobasket) and [iOS](https://apps.apple.com/us/app/vasto-basket/id6478608044).

We've built the entire data structure in-house, from scratch, to move quickly. Typical League, Season, Match, Team, Player, Staff, type thing.

The first season went great and the response was so big that now other teams from different sports want to be part of it so we need to handle a lot more different cases, especially different competition formats.
That's why I started looking for more general and concrete data solutions in the world of sports, and stumbled upon SportsML before finding this as the evolution of it.

The individual sports requirement came from a Table Tennis org, that needs to handle both singles and doubles competitions. The coaching staff will be mostly assigned to the single athletes.

Looking forward to knowing more after your meeting.

**Comment by pauljkelly** [comment]
Regarding club, looks like there's some demand for this. See [here](https://github.com/iptc/sport-schema/issues/193). Feel free to comment there. No problem adding it once we work through the nuances. 

Regarding the teams around individual athletes we thought a property on Associate might work that links straight to the athlete (rather than create a new relationship between classes). Let us know.

**Comment by Pupix** [comment]
I am not sure if a single property would be enough, because you'd still need `startDate`, `endDate`, and `positionRegular` off the top of my head. Right now I am using the `IndividualMembership` to represent this data, using `membershipOf` linking to an `Athlete` instead of a `Team`.

Added one more example to the clubs discussion and will stay on top of that as well.

**Comment by pauljkelly** [comment]
Yeah OK. So there's membershipOf in Membership. Would it be weird to point that to an Athlete? (It would be membership in the sense of "associate of")

**Comment by bquinn** [comment]
Current suggestion: add an entity that's a child of Membership (and so has "startdate" and "enddate") that points to Associate and Athlete.

<img width="708" alt="Screenshot 2024-08-06 at 16 41 36" src="https://github.com/user-attachments/assets/ef8253e1-bd13-44f8-b01b-9bfafa1104e8">

The big question is what to call it! Names we threw around were "Athlete Association" and more...

**Comment by Pupix** [comment]
Sorry, I've been on summer holiday during the past month.

Isn't the current implementation limited? Seems like a patch to handle this very specific use case between associate and athlete.

The associate can also have a *relationship* to a team, instead of the athlete. What then? Does it become an `IndividualMembership` then?

When I think of the association between a coach and his trainees, I think it's the same thing. So the coach either handles individual athletes or a whole team. The same thing events have. The competitors are either athletes or teams, grouped under `Competitor`.

The way I am reading it, if I change the trainee, I have to change the whole data structure instead of a *single field*.

Am I reading it properly?

**Comment by bquinn** [comment]
We discussed this today, here's our new approach:

![IPTC Sport Schema - Sport Schema v1 1 full model (1)](https://github.com/user-attachments/assets/190f3a1e-cc20-4ae4-b833-250b834e05c2)

**Comment by Pupix** [comment]
This looks a lot better and also solves the problem I pointed out about the `Associate` being able to change the *Trainee*.
With that being said I don't have anything else to add to this specific topic. I will follow along on the other issues and intervene when need be.

Feel free to close whenever. Thanks!
